app/components/game_board_component.rb

```diff
@@ -13,15 +13,27 @@ def initialize(game:, attempts:)
     @attempts = attempts
   end
 
+  def current_attempt_class(guess_number)
+    "current-attempt" if guess_number == current_attempt
+  end
+
   def guesses
     (1..guess_attempts).to_a.reverse
   end
 
   def guess_class(guess_number, cell_number)
+    classes = %w(guess-cell ba b--black dib ml3 mt2 tc br4 rc)
+
     attempt = attempts[guess_number - 1]
     guessed_value = attempt ? attempt.values[cell_number]  : nil
     bg_class = guessed_value ? "bg-#{guessed_value}" : "bg-black"
-    "guess-cell ba b--black dib ml3 mt2 tc br4 rc #{bg_class}"
+    classes << bg_class
+
+    if guess_number == current_attempt && cell_number == 0
+      classes << "current-cell"
+    end
```

Comment:
I am a bit unclear about why the cell_number has to be 0 alone before the current cell can be added to classes (line 32)

Reply:
When the server renders the HTML for the page, so far, it should always start with the first of the 4 cells being marked as _current_. Each cell essentially has 2 coordinates: row number as `guess_number` & column number as `cell_number`.

Reply:
Dealing with calculating the next 3 cells is left for the javascript. Javascript only needs to find the next sibling element, the row never changes in the lifetime of that small script. And they don't share the same logic or counting. Ruby generates correct HTML. That HTML is entirely sufficient for the Javascript part to figure out its own responsibilities, that is figuring out the right elements & changing their classes.

Reply:
Ooh, got the point now. Thank you so much for the further explanation.